joblib/test/test_parallel.py

```diff
@@ -1026,6 +1045,27 @@ def test_backend_batch_statistics_reset(backend):
     assert test_time / ref_time <= 1 + relative_tolerance
 
 
+@with_numpy
+@parametrize('backend', BACKENDS)
+@parametrize('n_jobs', [1, 2, -2, -1])
+def test_deadlock_with_generator(backend, n_jobs):
+    # This test used to timeout with multiprocessing backend.
+    # We use a large numpy object to delay the pickler, which causes Pipes
+    # to timeout.
```

Comment:
You should make it explicitly that this is a non-regression test for a race condition in the backends. I find the current phrasing confusing.